src/btree/bt_vrfy.c

```diff
@@ -280,12 +284,31 @@ __verify_key_hs(WT_SESSION_IMPL *session, WT_ITEM *key, WT_VSTUFF *vs)
 #else
         WT_UNUSED(vs);
 #endif
-    }
 
+        /* Perform Verification of data continuity */
+        if (orig_start.timestamp < stop.timestamp) {
+            ts1_bp = __verify_timestamp_to_pretty_string(stop.timestamp);
+            ts2_bp = __verify_timestamp_to_pretty_string(orig_start.timestamp);
+            WT_ERR_MSG(session, WT_ERROR,
+              "Key %s has a history store stop timestamp %s newer than its start timestamp %s",
+              __wt_buf_set_printable(session, hs_key->data, hs_key->size, vs->tmp1), ts1_bp,
+              ts2_bp);
+        }
+        if (orig_start.txnid < stop.txnid) {
+            WT_ERR_MSG(session, WT_ERROR, "Key %s has a history store stop transaction (%" PRIu64
+                                          ") newer than its start transaction (%" PRIu64 ")",
+              __wt_buf_set_printable(session, hs_key->data, hs_key->size, vs->tmp1), stop.txnid,
+              orig_start.txnid);
+        }
+        memcpy(&orig_start, &start, sizeof(WT_TIME_PAIR));
```

Comment:
could just assign values directly like when you initialized?

Reply:
agreed.